### ISSUE TITLE
[css-overflow-5] :focus and :focus-within styles with focused scroller pseudo-element

### ISSUE THREAD

**Issue by flackr** [w3c/csswg-drafts#11361]
When the user has focused a `::scroll-button` or `::scroll-marker`, the returned `document.activeElement` is the scrolling container as the activeElement API expects an element and this is consistent with having an "inner navigable" on the scrolling container similar to a shadow tree or inner frame per https://html.spec.whatwg.org/multipage/interaction.html#dom-documentorshadowroot-activeelement-dev:

> For the purposes of this API, when a [child navigable](https://html.spec.whatwg.org/multipage/document-sequences.html#child-navigable) is focused, its [container](https://html.spec.whatwg.org/multipage/document-sequences.html#nav-container) is [focused](https://html.spec.whatwg.org/multipage/interaction.html#bc-focus-ergo-bcc-focus) within its [parent](https://html.spec.whatwg.org/multipage/document-sequences.html#nav-parent)'s [active document](https://html.spec.whatwg.org/multipage/document-sequences.html#nav-document). For example, if the user moves the focus to a text control in an [iframe](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element), the [iframe](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element) is the element returned by the [activeElement](https://html.spec.whatwg.org/multipage/interaction.html#dom-documentorshadowroot-activeelement) API in the [iframe](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element)'s [node document](https://dom.spec.whatwg.org/#concept-node-document).

Similarly, when the focused element is in a different [node tree](https://dom.spec.whatwg.org/#concept-node-tree) than documentOrShadowRoot, the element returned will be the [host](https://dom.spec.whatwg.org/#concept-documentfragment-host) that's located in the same [node tree](https://dom.spec.whatwg.org/#concept-node-tree) as documentOrShadowRoot if documentOrShadowRoot is a [shadow-including inclusive ancestor](https://dom.spec.whatwg.org/#concept-shadow-including-inclusive-ancestor) of the focused element, and null if not.

The question is when these pseudos are focused, which elements have `:focus` or `:focus-within` styles?

Since the pseudos themselves can be styled by `:focus`, I think the most natural would be that the scrolling container matches `:focus-within` but not `:focus`. This seems the most natural, but is inconsistent with iframes and shadow dom (see [demo](https://codepen.io/flackr/pen/rNgVpOq)):
* When an element in a subframe is focused, it seems to not apply focus or focus-within to the frame container
* shadow dom applies `:focus` to the host element and `:focus-within` to it and its ancestors.

**Comment by tabatkins** [comment]
I agree with this, and think the "inconsistency" is just fine. Despite some similarities, pseudo-element trees aren't shadow DOMs or iframes; they more closely resemble ordinary light-dom subtrees in the relevant aspects here.

(iframes are independent documents, and so can each have a focused element; reflecting their focus into the parent would imply the parent document can have multiple focused elements, violating an invariant we currently hold to. Shadow trees don't expose their existence at all in many ways, and the outside world instead sees *only* the host element; the host thus can't be :focus-within without exposing the existence of a shadow tree that holds the :focus.)

So, the focused pseudo-element should match :focus, and when a scroll-marker is focused, the scroll-marker-group should match :focus-within.
